### PR TITLE
Centralise test imports and add os(Android)

### PR DESCRIPTION
**Pull request by johnno1962**
Hi Apple,

This PR adds os(Android) to the conditional imports and centralises them to reduce some boilerplate.

### CODE REVIEW

**Comment by parkera** [comment]
@swift-ci please test

File: TestFoundation/main.swift [hunk 1]
```diff
@@ -7,21 +7,14 @@
...
-import SwiftXCTest
-#endif
+// Most imports now centraized in TestImports.swift
 
 #if os(macOS) || os(iOS)
     import Darwin
-#elseif os(Linux)
+#elseif os(Linux) || os(Android)
     import Glibc
 #endif
```
Comment (spevans) [inline TestFoundation/main.swift]: Is it worth simplifying this to:
```
#if canImport(Darwin)
    import Darwin
#elseif canImport(Glibc)
    import (Glibc)
#endif
```

(and also in `HTTPServer.swift`)

**Comment by johnno1962** [comment]
The test failure should be fixed now.

**Comment by alblue** [comment]
Seems to be a failure in something otherwise unrelated:

```
13:57:57 
TestFoundation/XDGTestHelper.swift:13:23: error: use of unresolved identifier 'HTTPCookieStorage'
13:57:57         let storage = HTTPCookieStorage.shared
13:57:57                       ^~~~~~~~~~~~~~~~~
13:57:57 TestFoundation/XDGTestHelper.swift:14:26: error: use of undeclared type 'HTTPCookiePropertyKey'
13:57:57         let properties: [HTTPCookiePropertyKey: String] = [
13:57:57                          ^~~~~~~~~~~~~~~~~~~~~
13:57:57 TestFoundation/XDGTestHelper.swift:20:34: error: use of unresolved identifier 'HTTPCookie'
13:57:57         guard let simpleCookie = HTTPCookie(properties: properties) else {
13:57:57                                  ^~~~~~~~~~
```

**Comment by alblue** [comment]
@swift-ci please test

**Comment by parkera** [comment]
@johnno1962 can you also confirm this works when running the tests in Xcode on Darwin?

**Comment by johnno1962** [comment]
I wish I could but I’m still running Xcode 9.3 and can’t get past Foundation. Downloading the beta now..

**Comment by millenomi** [comment]
@johnno1962 Make sure you're using a current Swift compiler toolchain (either built with ninja, or via a swift.org snapshot.)

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by johnno1962** [comment]
I’m up to date now using master from swift.org. I’m seeing a seemingly unrelated crash through on macOS here: https://github.com/apple/swift-corelibs-foundation/blob/master/CoreFoundation/String.subproj/CFStringEncodings.c#L639

**Comment by spevans** [comment]
@johnno1962 See https://github.com/apple/swift-corelibs-foundation/pull/1498 for details of that crash.

**Comment by millenomi** [comment]
@swift-ci please test and merge

**Comment by johnno1962** [comment]
Thanks @millenomi

**Pull request merged**